test/tests/functional/pbs_svr_dyn_res.py

```diff
@@ -150,27 +185,27 @@ def test_res_whitespace(self):
         """
         # Create a resource of type long
         resname = ["foo"]
-        restype = "long"
+        restype = ["long"]
 
         # Prep for server_dyn_resource scripts. Script "PbsPtl_get_foo*"
         # generates file "PbsPtl_got_foo" and returns 1.
-        script_body = "echo get_foo > /tmp/PbsPtl_got_foo; echo 1"
+        script_body = "echo get_foo > /tmp/PtlPbs_got_foo; echo 1"
 
-        fpath_out = os.path.join(os.sep, "tmp", "PbsPtl_got_foo")
+        fpath_out = os.path.join(os.sep, "tmp", "PtlPbs_got_foo")
 
         (fd_in, fn_in) = self.du.mkstemp(prefix="PtlPbs_get_foo",
                                          body=script_body)
         self.du.chmod(path=fn_in, mode=0755, sudo=True)
         os.close(fd_in)
 
         # Add additional white space between resource name and the script
-        resval = ['"' + resname[0] + ' !' + fn_in + '"']
+        resval = ['"' + resname[0] + '  ' + ' !' + fn_in + '"']
 
         self.setup_dyn_res(resname, restype, resval)
 
         # Check if the file "PbsPtl_got_foo" was created
-        time.sleep(15)
-        self.assertTrue(self.du.isfile(path=fpath_out, sudo=True))
+        time.sleep(5)
```

Comment:
instead of sleep please add a loop which loops every 1s for may be say 10s till the file appeared.